nyaml.mk

```diff
@@ -0,0 +1,22 @@
+BASE_CLASS_DIR := base_classes
+CONTRIB_DIR := contributed_definitions
+APPDEF_DIR := applications
+NYAML_SUBDIR := nyaml
+NXDL_BC = $(addprefix $(BASE_CLASS_DIR)/nyaml/,$(notdir $(patsubst %.nxdl.xml,%.yaml,$(wildcard $(BASE_CLASS_DIR)/*.nxdl.xml))))
+NXDL_APPDEF = $(addprefix $(APPDEF_DIR)/nyaml/,$(notdir $(patsubst %.nxdl.xml,%.yaml,$(wildcard $(APPDEF_DIR)/*.nxdl.xml))))
+NXDL_CONTRIB = $(addprefix $(CONTRIB_DIR)/nyaml/,$(notdir $(patsubst %.nxdl.xml,%.yaml,$(wildcard $(CONTRIB_DIR)/*.nxdl.xml))))
+
+.PHONY: all nyaml
+
+$(BASE_CLASS_DIR)/$(NYAML_SUBDIR)/%.yaml : $(BASE_CLASS_DIR)/%.nxdl.xml
+	nyaml2nxdl $< --output-file $@
+
+$(CONTRIB_DIR)/$(NYAML_SUBDIR)/%.yaml : $(CONTRIB_DIR)/%.nxdl.xml
+	nyaml2nxdl $< --output-file $@
+
+$(APPDEF_DIR)/$(NYAML_SUBDIR)/%.yaml : $(APPDEF_DIR)/%.nxdl.xml
+	nyaml2nxdl $< --output-file $@
+
+nyaml: $(NXDL_BC) $(NXDL_APPDEF) $(NXDL_CONTRIB)
```

Comment:
why does it come to a separate makefile?

Reply:
That way I can avoid the cyclic dependency between building yaml and nxdl files and we don't have to do this for-loop based build anymore. This also allows building a single yaml file or only files which changed. I don't know if there is a better way of including this in a single file (e.g., namespacing the different rules so that there is no cyclic dependency anymore).

I guess it's up for discussion whether we prefer this approach or the old for-loop based approach without the second file. Both would work.

Reply:
I checked and they are passed down.